### DISCUSSION TITLE
Any reason the template HRESULT members are internal?

### DISCUSSION THREAD

**Discussion by AliveDevil** [microsoft/CsWin32#456] [Q&A] [answered]
I don't see any benefit of internalizing ThrowOnFailure: https://github.com/microsoft/CsWin32/blob/main/src/Microsoft.Windows.CsWin32/templates/HRESULT.cs#L43-L62

Same with `Succeeded` and `Failed`, consuming these members in any dependent project is impossible, unless `InternalsVisibleTo` is set.

**Comment by AArnott** [comment, marked as answer]
Thank you for calling this out. This is an oversight. By default all generated types are `internal`, but even when the option to generate public types is set, these template-supplied members remain internal. We'll get this fixed.

**Reply by AArnott** [reply]
See #457
